### PR TITLE
v0.0.13

### LINKED ISSUE #62: Allow RP1210Client and RP1210VendorList to initialize from alternate directories

### LINKED ISSUE #70: Pass Protocol arg in ClientConnect() through sanitize_msg_param()
This will allow regular strings to be used as well as bytestrings.

### LINKED ISSUE #71: Pass ClientMessage arg in SendMessage() through sanitize_msg_param()
This is required to make it compatible with the J1939Message class (which will evaluate to bytes in sanitize_msg_param()).

### LINKED ISSUE #75: RP1210API.GetErrorMsg should fall back to translating the error message rather than translating GetErrorMsg's return code.
When a user calls GetErrorMsg, they want a translation of their error code. If RP1210_GetErrorMsg returns a code, we don't want to translate that instead of ErrorMessage, since that's confusing.

Instead of:
```python
return translateErrorCode(ret_code)
```
We should have:
```python
return translateErrorCode(ErrorCode)
```

### LINKED ISSUE #76: Translate the return value of GetHardwareStatusDirect into a string
replace
```python
return ClientInfo
```
with
```python
return str(ClientInfo.value, "utf-8")
```

### PR DESCRIPTION
**Pull request by dfieschko**
- Pass relevant args of API functions in RP1210API through sanitize_msg_param()
   1. ClientMessage arg in SendMessage()
   2. Protocol arg in ClientConnect()
- Add ReadDLLVersion() and ReadAPIVersion() to RP1210API class
- Add pull request template
- Update some docstrings & type hints
- Add DrewLinQ drivers to automated unit tests
   - DrewLinQ.ini/dll
   - DTKRP32.ini/dll
- RP1210API.GetHardwareStatus() will now automatically determine buffer size by default
- Handle case in getProtocol() when user inputs value that isn't int or str
- RP1210VendorList update
   - Add ability to specify alternate paths for RP121032.ini, INI directory, and DLL directory
   - Add __getitem__(), which provides direct access to vendors
   - Add __bool__(), which returns True if vendors were populated, False if not
   - Add __len__(), which returns number of vendors stored in list
- Re-enable and refactor RP1210VendorList tests
   - Bumped code coverage to 92%
- Fall back to translating ErrorCode in GetErrorMsg instead of returning ret_val
- Translate the return value of GetHardwareStatusDirect into a string

closes #70, closes #71, closes #62, closes #75, closes #76

### CODE REVIEW

File: RP1210/RP1210.py [hunk 1]
```diff
@@ -1159,24 +1179,33 @@ class RP1210VendorList:
     - Set device index with `setDeviceIndex()`. This is NOT deviceID!
     - If you have a vendor name but not index, use `getVendorIndex(api_name)` to find the index.
     """
-    def __init__(self):
-        super().__init__()
+    def __init__(self, rp121032_path : str = None, api_dir : str = None, config_dir : str = None):
```
[the remark this answers is not in the document]
Comment (dfieschko) [reply RP1210/RP1210.py:1182]: @sonatype-lift ignore

File: RP1210/RP1210.py [hunk 2]
```diff
@@ -1268,20 +1300,24 @@ def getDeviceIndex(self, deviceID = -1) -> int:
             return 0
         return 0
 
-    def getVendor(self, index : int) -> RP1210Config:
+    def getVendor(self, index : int = None) -> RP1210Config:
```
[the remark this answers is not in the document]
Comment (dfieschko) [reply RP1210/RP1210.py:1303]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (dfieschko) [reply RP1210/RP1210.py:1303]: Really wish it wouldn't trigger on `= None`

**Comment by lisi12345** [comment]
Made some updates for issues #75 and #76

1. Fall back to translating ErrorCode in GetErrorMsg instead of returning ret_val
2. Translate the return value of GetHardwareStatusDirect into a string

File: RP1210/RP1210.py [hunk 3]
```diff
@@ -1064,31 +1083,31 @@ def GetErrorMsg(self, ErrorCode : int) -> str:
...
-        InfoSize must be 16 <= InfoSize <= 64, and must be a multiple of 2. Leave InfoSize blank
-        to default to 64.
+        BufferSize must range between 16 and 64, and must be a multiple of 2 (defaults to 64).
         """
-        ClientInfo = create_string_buffer(InfoSize)
-        self.getDLL().RP1210_GetHardwareStatus(ClientID, ClientInfo, InfoSize, 0)
-        return ClientInfo
+        ClientInfo = create_string_buffer(BufferSize)
+        self.getDLL().RP1210_GetHardwareStatus(ClientID, ClientInfo, BufferSize, 0)
+        return str(ClientInfo.value, "utf-8")
```
[the remark this answers is not in the document]
Comment (lisi12345) [reply RP1210/RP1210.py:1110]: Fixed with commit

**Pull request merged**